tests/Unit/Adapter/Shop/MaintenanceConfigurationTest.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Unit\Adapter\Preferences;
+
+use PrestaShop\PrestaShop\Adapter\Shop\MaintenanceConfiguration;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+use Tests\TestCase\AbstractConfigurationTestCase;
+
+class MaintenanceConfigurationTest extends AbstractConfigurationTestCase
+{
+    private const SHOP_ID = 42;
+
+    /**
+     * @dataProvider provideShopConstraints
+     *
+     * @param ShopConstraint $shopConstraint
+     */
+    public function testGetConfiguration(ShopConstraint $shopConstraint): void
```

Comment:
Ideally the mock should be built based on various shop constraints which return different values (so that you can check the difference). So the input configuration of the test would be like:

```php
[
    // This is the value returned by getShopConstraint, and also the parameter accepted by get
    ShopConstraint::shop(self::SHOP_ID),
    // This is the value returned by get (if the input constraint is correct, you can use the with function of the mock to assert this)
    [
        'PS_SHOP_ENABLE' => true,
        'PS_MAINTENANCE_IP' => 'test',
        'PS_MAINTENANCE_TEXT' => 'test',
    ],
    // This is the expected value asserted against the return of getConfiguration
    [
        'enable_shop' => true,
        'maintenance_ip' => 'test',
        'maintenance_text' => 'test',
    ],
]
```

Reply:
The construction of the two mocks could be placed into the `AbstractConfigurationTestCase` since they will be built the same way (assert and handle the association between a ShopConstraint input and a returned array)